jax/_src/lax/control_flow.py

```diff
@@ -1029,7 +1029,7 @@ def transposed(*args):
     res, cts_out = split_list(args, [num_res])
     primals = res + [ad.UndefinedPrimal(aval) for aval in primal_avals]
     cts_in = ad.backward_pass(
-        jaxpr.jaxpr, jaxpr.consts, primals, cts_out)
+        jaxpr.jaxpr, (), jaxpr.consts, primals, cts_out)
```

Comment:
It isn't "wrong" in the sense that it's always legal to delay reduction until later (in this case, until after control flow blocks), but performing reductions eagerly is the whole point of this PR so I went ahead and added that "optimization" to control flow as well rather than leaving that to another change.

(It's only legal to move reductions around if the transpose is actually linear; in principle this can be violated by custom VJP/custom transpose functions that don't implement the true VJP/transpose).